mutation.hh

```diff
@@ -151,19 +152,37 @@ private:
 namespace {
 
 template<consume_in_reverse reverse, FlattenedConsumer Consumer>
-stop_iteration consume_clustering_fragments(const schema& s, mutation_partition& partition, Consumer& consumer) {
+stop_iteration consume_clustering_fragments(schema_ptr s, mutation_partition& partition, Consumer& consumer) {
     using crs_type = mutation_partition::rows_type;
-    using crs_iterator_type = std::conditional_t<reverse == consume_in_reverse::yes, crs_type::reverse_iterator, crs_type::iterator>;
+    using crs_iterator_type = std::conditional_t<reverse == consume_in_reverse::legacy_half_reverse || reverse == consume_in_reverse::yes, crs_type::reverse_iterator, crs_type::iterator>;
     using rts_type = range_tombstone_list;
-    using rts_iterator_type = std::conditional_t<reverse == consume_in_reverse::yes, rts_type::reverse_iterator, rts_type::iterator>;
+    using rts_iterator_type = std::conditional_t<reverse == consume_in_reverse::legacy_half_reverse, rts_type::reverse_iterator, rts_type::iterator>;
+
+    if constexpr (reverse == consume_in_reverse::yes) {
+        s = s->make_reversed();
```

Comment:
The `mutation` doesn't know and doesn't care. Reverse is relative to the schema the mutation was produced with, which might have been a forward or reverse schema, it doesn't matter.

Reply:
I meant in relative terms.
Is `s` passed in already reversed (compared to the mutation), so we unreverse it, or is it "forward" (i.e. the same as mutation), in which case we reverse it here?

Reply:
I think the `if` just above makes this unambiguous: when we read in reverse we reverse the schema so it is the reverse of whatever the mutation has.

Reply:
Or the passed-in schema may already be the reverse of the mutation, then this code unreverses it back because it needs the original mutation schema for whatever reason.

It's not obvious which schema is passed through the interface.

For example, `mx::make_reader` on the sstable reader PR expects the already-reversed schema passed in, it doesn't do the reversing itself. Same happens on the memtable reader PR, and the memtable reader PR sometimes needs the original schema too, so it calls `make_reversed` (which then unreverses the schema).

Reply:
This is a private method used by `mutation::consume()` and it *always* passes in the mutation's schema unchanged.